feg/gateway/services/n7_n40_proxy/n7/n7_create_test.go

```diff
@@ -0,0 +1,233 @@
+		},
+		RatSpecificContext: &protos.RatSpecificContext{
+			Context: &protos.RatSpecificContext_M5GsmSessionContext{
+				M5GsmSessionContext: &protos.M5GSMSessionContext{
+					PduSessionId:   uint32(PDU_SESSION_ID),
+					Gpsi:           GPSI1,
+					PduSessionType: protos.PduSessionType_IPV4,
+				},
+			},
+		},
+		SessionId:      genSessionID(IMSI1),
+		AccessTimezone: &protos.Timezone{OffsetMinutes: 330},
+	}
+	expected := &n7_sbi.PostSmPoliciesJSONRequestBody{
+		AccessType:      &AccessType3gpp,
+		Dnn:             DnnMagma,
+		Gpsi:            &Gpsi1,
+		Ipv4Address:     &UeIpv4,
+		PduSessionId:    sbi.PduSessionId(PDU_SESSION_ID),
+		PduSessionType:  PduSessionTypeIpv4,
+		RatType:         &RatTypeNR,
+		Supi:            sbi.Supi(IMSI1),
+		UeTimeZone:      &UeTzIST,
+		NotificationUri: n7.GenNotifyUrl(NOTIFY_API_ROOT, genSessionID(IMSI1)),
+	}
+
+	reqBody := n7.GetSmPolicyContextDataN7(csrProto, NOTIFY_API_ROOT)
+	// Check if JSON conversion is successufl
+	_, err := json.Marshal(reqBody)
+	require.NoError(t, err)
+	assert.Equal(t, expected, reqBody)
+}
+
+func TestCreateSessionResponseProto(t *testing.T) {
+	policyDecisionStr := `{
+		"pccRules": {
+			"rule1": {
+				"pccRuleId": "rule1",
+				"flowInfos": [{
+					"flowDescription": "permit in ip from 0.0.0.0/0 to 4.2.2.4"
+				}],
+				"precedence": 1,
+				"refQosData": ["qos_data1"],
+				"refTcData": ["tc_data1"],
+				"refChgData": ["chg_data1"],
+				"refUmData": ["um_data1"],
+				"refCondData": "cond_data1"
+			},
+			"static_rule1": {
+				"pccRuleId": "static_rule1",
+				"refCondData": "cond_data1"
+			}
+		},
+		"qosDesc": {
+			"qos_data1": {
+				"qodId": "qos_data1",
+				"5qi": 5,
+				"maxbrUl": "200000",
+				"maxbrDl": "500000",
+				"gbrUl": "100000",
+				"gbrDl": "250000"
+			}
+		},
+		"chgDecs": {
+			"chg_data1": {
+				"chgId": "chg_data1",
+				"online": true,
+				"ratingGroup": 1,
+				"serviceId": 12
+			}
+		},
+		"traffContDecs": {
+			"tc_data1": {
+				"tcId": "tc_data1",
+				"redirectInfo": {
+					"redirectEnabled": true,
+					"redirectAddressType": "URL",
+					"redirectServerAddress": "https://redirect.example.com/tc"
+				}
+			}
+		},
+		"umDecs": {
+			"um_data1": {
+				"umId": "um_data1",
+				"volumeThreshold": 4000000,
+				"volumeThresholdUplink": 1500000,
+				"volumeThresholdDownlink": 3500000
+			}
+		},
+		"conds": {
+			"cond_data1": {
+				"condId": "cond_data1",
+				"activationTime": "2021-10-22T12:42:31Z",
+				"deactivationTime": "2021-10-22T14:42:31Z"
+			}
+		},
+		"policyCtrlReqTriggers": ["RE_TIMEOUT"],
+		"revalidationTime": "2021-10-22T14:42:31Z",
+		"online": true
+	}`
+	// Unmarshal json to openapi struct
+	var policyDecision n7_sbi.SmPolicyDecision
+	err := json.Unmarshal([]byte(policyDecisionStr), &policyDecision)
+	require.NoError(t, err)
+
+	csrProto := &protos.CreateSessionRequest{
+		CommonContext: &protos.CommonSessionContext{
+			Sid: &protos.SubscriberID{Id: IMSI1},
+		},
+		SessionId: genSessionID(IMSI1),
```

Comment:
do you really need to use this function?  Can't not you use a constant ?
Better to have tests with deterministic values (even that you may have find otherwise in session proxy unfortunately)